Object's Storage Dashboard (#219)
openshift/openshift-origin-design: web-console/4.0-designs/storage/object-storage-dashboard/object-storage-dashboard.md

# Object Storage Dashboard￼

> ## Vision

Reviewer: Should we add some details here about when it will appear in the Console? Will it only show if the OCS operator is installed?

Reviewer 2: @yuvalgalanti 

Let's have the following text for what the Vision is:
The main goal of the Object Service Dashboard is to provide an at-a-glance view of the Multi-Cloud Gateway (MCG) configuration, that is deployed as part of OpenShift Container Storage (OCS).  This dashboard would provide health and status information as well as reporting on key performance indicators for analysis and decision making.

Our main goal was to have an object's storage dashboard supported by MCG (NooBaa) that provide the user with at-a-glance information used for analysis and decision making. 

## Object Dashboards

![Object Storage Dashboard](img/object-storage-dashboard.png)

**The objects dashboard includes the following cards:**
- Details
- Buckets (Object buckets and Object bucket claims)
- Resources providers
- Health
- Data consumption 
- Data resiliency 
- Usage breakdown
- Data Reduction 

## Details Card
![Details Crad](img/details-card.png)

**The details card shows:**
- connected service name and offer external link to the NooBaa (MCG) system. 
- Provider name
- MCG installed version

## Buckets Card 

> ![Buckets Crad](img/buckets-card.png)

Reviewer 3: @julienlim no quota for 4.2. Unless you are talking about 4.3 or further

> ![Buckets Crad](img/buckets-card.png)

Reviewer 2: Ack - OBC quotas are out of scope for OCS 4.2.  No need to address after all.

MCG - MCG/NooBaa Buckets are data containers that behave similarly to AWS S3 buckets. Users can set policies that will determine which resources will be user (per bucket) to store data.

> We will show a breakdown between static buckets and buckets claim. Each bucket type will have a status indication when something is wrong with X buckets (shows the number of buckets that are currently in warning/ error status). The number link should lead to the problematic buckets list. 

Reviewer 4: Right now the "Down arrow" icon with the blue link doesn't look like a "Problem".
Should that link be red? maybe a label?

Reviewer: @mceledonia @andybraren - Thoughts on this from a consistency standpoint with the Overview Dashboard?

Reviewer 5: @yuvalgalanti IIRC we discussed changing these icons to the regular error/warning icons, right? I don't know anything about buckets, but even if "down" is what the status is technically called, an error icon with "Down" as the label would probably be clearer here. I would expect the same icon and label to be used in a future Buckets List page (if that page already exists, the "12 Object Buckets" label should also be blue here).

## Resources Providers Card

> ![Resources Providers](img/resource-card.png)

Reviewer 2: @shirimordechay @yuvalgalanti 

We need to change the "Ceph" mentions in this card and others to "OCS."

> ![Resources Providers](img/resource-card.png)

Reviewer 3: @julienlim isn't writing OCS creates a recursive loop? What happens in 4.3 when the ceph cluster is external?

> ![Resources Providers](img/resource-card.png)

Reviewer 6: @yuvalgalanti Why changing Cepg to OCS? it doesn't make sense. Both Ceph and MCG are a part of OCS.

> ![Resources Providers](img/resource-card.png)

Author: +1 I think it can be confusing to call it OCS because of OCS in my eyes is the entire storage "world" when its installed and MCG and Ceph are part of that

> ![Resources Providers](img/resource-card.png)

Reviewer 2: The reason I am suggesting to say OCS vs. Ceph is that it's going to be backed by OCS, and a lot of the references are OCS in other places, and it may get confusing to the user which one it is.  Also, in the future when we add RHCS support (external cluster), that's what most folks will think of as Ceph and not OCS.

@shirimordechay @yuvalgalanti @nimrod-becker

> This card list the different storage resources the user have. It can be a cloud based resource and a bare metal resource. For each type (AWS, google, ceph rgw…) we will show the number of resources and their status indication if something is wrong.

Reviewer: I think an extra space is needed before this line. Looks like the text is starting right after the image in the preview.

**Only used resources types will be shown, no need to show types that are not in used. 

> ## Health Card

Reviewer 6: @yuvalgalanti Please add the following texts to the PR:

**Info icon texts:**
**Resources Providers:**
A list of all MCG (Multi-cloud gateway) resources that are currently in use. Those resources are used to store data according to the buckets policies and can be a cloud-based resource or a bare metal resource. 

**Health:**
The object storage health state consists of the system’s data availability, buckets, and resources health and functionality.

**Data resiliency:** 
Data resiliency is the ability of stored objects to recover and continue operating in the case of a failure. 

Certain changes in the system (unavailable resource/ change of bucket policy etc.) cause an object to require a rebuilding process in order to stay resilient. 

**Efficiency ratio:**
Efficiency ratio refers to the deduplication and compression process effectiveness. 

**Savings:**
Savings shows the uncompressed and non-deduped data that would have been stored without those techniques. 

"See All" Link should show a subsystem Health card (such as the cluster dashboard) assuming it can be implemented in 4.2 
![image](https://user-images.githubusercontent.com/10331355/60430364-cf684f00-9c05-11e9-859c-5b2ea737b3d9.png)

If not possible, we can remove the "See all" link for now

All other "view x" should link to the MCG UI in its relevant page.

> ## Health Card

Author: Removing it for 4.2 sounds like a good idea. @ julienlim @etamir

> ## Health Card

Reviewer 3: If it's supported, why should we remove? (if not, of course, we'll remove)

![Health and alerts](img/health-card.png)

The health indication shows if the system is up and running without any major issues. When the system is in a warning or error state the alerts section will be shown and the relevant alerts will be displayed there. 

## Data Consumption Card

> In this card the user can see an analysis for providers and accounts (MCG accounts) by different KPIs (physical Usage, logical usage, i/o, and egress depends on the selected type). 

Reviewer: ```suggestion
In this card, the user can see an analysis for providers and MCG accounts by different KPIs (physical Usage, logical usage, i/o, and egress which are available selections based on KPI). 
```

### Accounts> I/O operations

> ![Accounts by I/O operations](img/data-consumption-crad-01.png)

Reviewer 3: @julienlim see my comment above

> ![Accounts by I/O operations](img/data-consumption-crad-01.png)

Reviewer 2: @nimrod-becker @shirimordechay @yuvalgalanti See comments above.

Show the top 5 accounts consumers R/W count values. At the bottom legends show the total Reads and total Writes amount of all 5. 
Why? States how connected applications utilizes the system resources. 

### Accounts > Logical Usage
![Accounts by Logical Usage](img/data-consumption-crad-02.png)

Show each account total logical usage size
Why? Show how connected applications utilizes the system resources. 

### Providers> I/O operations
![Providers by I/O operations](img/data-consumption-crad-04.png)

Input/Output operations sent to and from MCG to connected cloud providers
Why? What’s eating my budget? Understanding the traffic in the cloud, allowing the user to compare with the cloud billing reports, optimize and improve resources allocation according to the I/O pattern and perhaps consider changing the selected resource type.

### Providers> Egress
![Providers by Egress](img/data-consumption-crad-05.png)

> The size of the data NooBaa read from each provider 

Reviewer: MCG rather than NooBaa?

Reviewer 2: +1 on saying MCG and not NooBaa.

Why? Cost analysis.  Understanding the traffic in the cloud, allowing the user to compare with the cloud billing reports, optimize and improve resources allocation according to the egress pattern and perhaps consider changing the selected resource type.
    
### Providers> Physical vs. logical usage
![Providers by Physical vs. logical usage](img/data-consumption-crad-03.png)

Show the data consumption in the system by comparing the raw usage (physical) vs. the logical usage. Raw usage refers to the actual size that the buckets in the system are utilizing from it's resources including data resiliency replicas or fragments. 
Why? Allows the user to control storage efficiency and allocate appropriate resources according to usage characteristics.

## Data Resiliency Card

> ![Data Resiliency](img/data-resiliency-crad.png)

Reviewer: Is it worth showing the big green checkmark version of this too?

In this card we will show if there is any resiliency issue in MCG/Noobaa. For any rebuilding process we will show a progress bar that indicates the progression. Time estimation will be shown as well if available.

## Capacity Usage Card
![Capacity Usage Breakdown](img/capacity-usage-crad.png)

> Show a breakdown by pie chart of the usage of the object storage by projects/ Bucket Class.

Reviewer: I think an extra space is needed before this line. Looks like the text is starting right after the image in the preview.

## Data Reduction Card
![Data Reduction](img/data-reduction-crad.png)

> In this card we will show the data optimization in MCG which consist of deduplication and compression. 

Reviewer: It would be great to suggest the "info" text for these two items.
